packages/clay-layout/src/Content.tsx

```diff
@@ -0,0 +1,129 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import React from 'react';
+
+interface IRowProps extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * Element or component to render for container
+	 */
+	containerElement?: React.JSXElementConstructor<{
+		className: string;
+		[key: string]: any;
+	}>;
+
+	/*
+	 * Give negative margins on the top, right, bottom, and left to offset the padding
+	 */
+	noGutters?: 'x' | 'y' | true;
+
+	/*
+	 * Gives padding to all autofit-cols that are direct children of autofit-row.
+	 */
+	padded?: boolean;
+
+	/**
+	 * Adds class for aligning items within the row.
+	 */
+	verticalAlign?: 'center' | 'end';
+}
+
+const ContentRow: React.FunctionComponent<IRowProps> = ({
+	children,
+	className,
+	containerElement: ContainerElement = 'div',
+	noGutters,
+	padded,
+	verticalAlign,
+	...otherProps
+}) => {
+	return (
+		<ContainerElement
+			{...otherProps}
+			className={classNames(className, 'autofit-row', {
+				'autofit-padded': padded,
+				'autofit-padded-no-gutters': noGutters === true,
+				[`autofit-padded-no-gutters-${noGutters}`]:
+					typeof noGutters === 'string',
+				[`autofit-row-${verticalAlign}`]: verticalAlign,
+			})}
+		>
+			{children}
+		</ContainerElement>
+	);
+};
+
+ContentRow.displayName = 'ClayContentRow';
+
+interface IColProps extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * Element or component to render for container
+	 */
+	containerElement?: React.JSXElementConstructor<{
+		className: string;
+		[key: string]: any;
+	}>;
+
+	/*
+	 * Makes column expand and fill available space in row.
+	 */
+	expand?: boolean;
+
+	gutters?: boolean;
+}
+
+const ContentCol: React.FunctionComponent<IColProps> = ({
+	children,
+	className,
+	containerElement: ContainerElement = 'div',
+	expand,
+	gutters,
+	...otherProps
+}) => {
+	return (
+		<ContainerElement
+			{...otherProps}
+			className={classNames(className, 'autofit-col', {
+				'autofit-col-expand': expand,
```

Comment:
Looking at usages in DXP, `autofit-col-end` is also possible and very extended. It seems to be we usually have either nothing, `expand` or `end`. The `gutters` value is compatible with the above. 

We might want to add `end` as another prop... or maybe there's a better more semantic name for this...

@pat270, @marcoscv-work?  What's the difference between nothing, `expand` and `end`? How would we need to name an enumeration of this?

Reply:
@jbalsas `expand` is a good name.

We shouldn't include `autofit-col-end`. It was deprecated at https://github.com/liferay/clay/issues/803. The replacement for it is using `autofit-float-end` on `autofit-row`.

---

```
<div class="autofit-float-end autofit-row">
  <div class="autofit-col autofit-col-expand">
    <div class="autofit-section">
      The quick brown fox jumped over the lazy dog.
    </div>
  </div>
  <div class="autofit-col">
    <div class="autofit-section">
      <button>Add</button>
    </div>
  </div>
</div>
```

If the `autofit-col`s don't fit on one line, the last item will break to new line and everything (`autofit-col`) will be aligned right.

---

```
<div class="autofit-float-end-{sm|md}-down autofit-row"></div>
```
At the breakpoint `{sm|md}`, if `autofit-col`s don't fit on one line, last item will break to new line and aligned right.

---

```
<div class="autofit-float autofit-row"></div>
```

If doesn't fit on one line, breaks to new line and aligned left.

---

```
<div class="autofit-float-{sm|md}-down autofit-row"></div>
```

At breakpoint `{sm|md}`, if doesn't fit on one line, breaks to new line and aligned left.

---

- `autofit-col` sizes the column to be as wide as its content.

- `autofit-col autofit-col-expand` forces the column to fill the remaining space. If there are multiple `autofit-col-expand`s then the column widths are supposed to be divided equally. This doesn't work in all situations right now. I think we can fix this.

We also have:

```
<ul class="card-page card-page-equal-height">
  <li class="card-page-item col-{breakpoint}-#"></li>
  <li class="card-page-item col-{breakpoint}-#"></li>
  <li class="card-page-item col-{breakpoint}-#"></li>
</ul>
```

I don't think we should include this use case here, but just getting it out. We also have a grid for cards `card-page` and `card-page card-page-equal-height`. You can see a demo at https://pat270.github.io/clay3-test-site/vclay-table-dd/content/test-card-type-template/.

Reply:
> It was deprecated at #803

That was like 2 years ago! 😂 We will need to discuss also how to deprecate things in claycss in the future :D 

@bryceosterhaus, that means if I understand correctly that `expand` on col is okay, but we're missing `float` or something like that on [ContentRow](https://github.com/liferay/clay/pull/3204/files#diff-3d9a224ed0f982f449f890b93b0b359aR47)?

Reply:
> You can see a demo at https://pat270.github.io/clay3-test-site/vclay-table-dd/content/test-card-type-template/.

(Off topic, but) there are some funny flickering on that page in Chrome 81:

![Screenshot 2020-05-18 -115816-2fAZXhEY](https://user-images.githubusercontent.com/7074/82200524-67932380-98ff-11ea-935d-7b515d9765ab.gif)

Takes a while to trigger it, but you see it when I move the mouse off of the rightmost card.

Reply:
Thanks @wincent, not sure when this started happening (I'm blaming a Chrome update on this). Any `transition-timing-function` other than `ease-in` or `ease-in-out` fixes this artifact.